### PR TITLE
doc(webpack): webpack example

### LINKED ISSUE #781: docs: write docs reflecting how to use yargs with webpack
version:  6.6.0

It look like yargs  cannot be used with webpack ( tested with latest webpack version 2.2)

Here is how to reproduce

 webconfig.pack.js

```javascript

var path = require("path");
module.exports = {

    target: "node",
    context: __dirname,
    node: {
        console: true,
        global: true,
        process: true,
        Buffer: true,
        setImmediate: true,
        __dirname: false,
        __filename: false,

    },
    entry: {
        main: path.resolve(__dirname,"bug.js"),
    },
    output: {
        path: path.resolve(__dirname),
        filename: "[name].dist.js"
    },
 };
 
```

bug.js
```javascript 
var yargs = require("yargs");
```

package.json
```json
{
  "name": "yargs_issue_with_web_pack",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "webpack": "./node_modules/.bin/webpack"
  },
  "author": "",
  "license": "ISC",
  "devDependencies": {
    "webpack": "^2.2.1"
  },
  "dependencies": {
    "yargs": "^6.6.0"
  }
}
```

commande:

```
$npm run webpack
```


Output is
```
WARNING in ./~/yargs/index.js
11:39-46 Critical dependency: require function is used in a way in which dependencies cannot be statically extracted

WARNING in ./~/yargs/yargs.js
248:33-40 Critical dependency: require function is used in a way in which dependencies cannot be statically extracted

WARNING in ./~/yargs/yargs.js
466:71-78 Critical dependency: require function is used in a way in which dependencies cannot be statically extracted

WARNING in ./~/yargs-parser/index.js
444:21-48 Critical dependency: the request of a dependency is an expression

WARNING in ./~/require-main-filename/index.js
2:25-32 Critical dependency: require function is used in a way in which dependencies cannot be statically extracted
```

running the webpacked scripts 
    $ node main.dist.js


produces
```
C:\projects\stuff\yargs_issue_with_web_pack\main.dist.js:662                                                                                    
  const argv = yargs(processArgs, cwd, !(function webpackMissingModule() { var e = new Error("Cannot find module \".\""); e.code = 'MODULE_NOT_FOUND';; thro
                                                                                                                                                        ^   
                                                                                                                                                            
Error: Cannot find module "."                                                                                                                               
    at webpackMissingModule (C:\projects\stuff\yargs_issue_with_web_pack\main.dist.js:662:84)                                                   
    at Argv (C:\projects\stuff\yargs_issue_with_web_pack\main.dist.js:662:163)                                                                  
    at Object.module.exports (C:\projects\stuff\yargs_issue_with_web_pack\main.dist.js:657:1)                                                   
    at __webpack_require__ (C:\projects\stuff\yargs_issue_with_web_pack\main.dist.js:20:30)                                                     
    at Object.<anonymous> (C:\projects\stuff\yargs_issue_with_web_pack\main.dist.js:11603:13)                                                   
    at __webpack_require__ (C:\projects\stuff\yargs_issue_with_web_pack\main.dist.js:20:30)                                                     
    at C:\projects\stuff\yargs_issue_with_web_pack\main.dist.js:66:18                                                                           
    at Object.<anonymous> (C:\projects\stuff\yargs_issue_with_web_pack\main.dist.js:69:10)                                                      
    at Module._compile (module.js:570:32)                                                                                                                   
    at Object.Module._extensions..js (module.js:579:10)                                                                                                     
                                                                                                                                                            
```

### PR DESCRIPTION
**Pull request by mleguen**
Usage example for webpack, both in javascript and typescript.

I stumbled myself on this issue in one of my projects a few days ago, and found it was about time to close #781 !

### CODE REVIEW

File: docs/webpack.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+```
+
+Create `webpack.config.js`:
+```js
+const path = require('path');
+const nodeExternals = require('webpack-node-externals');
+
+module.exports = {
+  entry: './src/index.js',
+  externals: [nodeExternals()],
```
Comment (mleguen) [inline docs/webpack.md]: Removing all node_modules from the pack just to get rid of yargs webpack issues is excessive.

I will update this next week.

**Comment by bcoe** [comment]
@mleguen neat, thanks for pulling this together.

File: docs/webpack.md (hunk 1 above)
Comment (mleguen) [reply docs/webpack.md]: Fixed

File: docs/webpack.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
+# Webpack usage examples
+
+## Install yargs
+
+Install yargs as a production dependency, as it will _not_ be included in the pack (not yet compatible):
```
Comment (mleguen) [inline docs/webpack.md]: Yet some people could _want_ to ignore webpack warnings and have it included in the pack.

I am adding this.

Comment (mleguen) [reply docs/webpack.md]: Fixed

**Comment by mleguen** [comment]
Ready for reviews!

File: docs/webpack.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+Create `src/index.js`:
+```js
+const yargs = require('yargs')
+
+console.log(yargs.parse())
+```
+
+Or for typescript users, `src/index.ts`:
+```ts
+import * as yargs from 'yargs';
```
Comment (bcoe) [inline docs/webpack.md]: I think we're going to want to update the TypeScript examples themselves to use this approach as well, we swapped to the following approach:

https://github.com/yargs/yargs/commit/b1b156a3eb4ddd6803fbbd56c611a77919293000

But it seems to be causing trouble for folks, see: 

https://github.com/yargs/yargs/issues/1455

I think we actually addressed #1439 elsewhere, by removing the deprecation warning.

Comment (mleguen) [reply docs/webpack.md]: #1455 is on my TODO list, once I'm done with #1452

**Pull request merged**